src/lobpcg/algorithm.rs

```diff
@@ -0,0 +1,658 @@
+use crate::{cholesky::*, eigh::*, norm::*, triangular::*};
+///! Locally Optimal Block Preconditioned Conjugated
+///!
+///! This module implements the Locally Optimal Block Preconditioned Conjugated (LOBPCG) algorithm,
+///which can be used as a solver for large symmetric eigenproblems.
+use crate::{LinalgError, Result};
+use ndarray::prelude::*;
+use ndarray::{concatenate, Data, ScalarOperand};
+use num_traits::{Float, NumCast};
+
+use std::iter::Sum;
+
+/// Find largest or smallest eigenvalues
+#[derive(Debug, Clone)]
+pub enum Order {
+    Largest,
+    Smallest,
+}
```

Comment:
Maybe this enum should be moved to the crate root so we can also use it in `sort_eig` and `sort_svd`.